### ISSUE TITLE
Compositor bug with terrain tiles example

### ISSUE THREAD

**Issue by mlshapiro** [creare-com/podpac#405]
**Description**

I am going back to use the terrain tiles, which seems to be working correctly, but the compositor is giving strange behavior

**Steps to Reproduce**
```python
import podpac
from podpac.datalib.terraintiles import TerrainTiles

lat = podpac.clinspace(40, 41, 100)
lon = podpac.clinspace(-79, -78, 100)
time = ['2020-03-05', '2020-03-06']
coords = podpac.Coordinates([lat, lon, time], dims=['lat', 'lon', 'time'])

terrain = TerrainTiles(zoom=6)
output = terrain.eval(coords.drop("time"))  # this seems to evaluate the first "source" over and over for the length of "sources"
output.plot()  # this just shows the first source evaluated on the coordinates of the whole region
```

**Expected Behavior**

I would expect all the sources to get evaluated and then composited.

**Observed Behavior**

- the `terrain.sources` attribute is set correctly with different sources
- if I debug the compositor (L241 of compositor.py - add `print(list(outputs))`) I see the same output over and over

**Comment by mpu-creare** [comment]
My guess would be caching.

**Comment by mlshapiro** [comment]
I was in the midst of typing that... it seems to be caching, but its not a traditional "clear the cache" problem. Its like it keeps grabbing the same terrain tiles node because the source is not differentiating it.

**Comment by mlshapiro** [comment]
OK fixed - it was because TerrainTiles was overriding the `source` attribute of Rasterio (and hence BaseFile). Pushing a fix to develop.

**Comment by mlshapiro** [comment]
Fixed in  6ad60774c9ec5818ae904e641e4e58d75acc8e90
